src/Events/BeamBatchTransactionCreated.php

```diff
@@ -5,20 +5,21 @@
 use Enjin\Platform\Beam\Channels\PlatformBeamChannel;
 use Enjin\Platform\Channels\PlatformAppChannel;
 use Enjin\Platform\Events\PlatformBroadcastEvent;
+use Illuminate\Database\Eloquent\Model;
 
 class BeamBatchTransactionCreated extends PlatformBroadcastEvent
 {
     /**
      * Creates a new event instance.
      */
-    public function __construct(string $beamId, string $collectionId, int $transactionId)
+    public function __construct(mixed $event, ?Model $transaction, ?array $extra)
```

Comment:
Mostly there's no data passed to `$transaction` and `$extra` better put default to `null`

Reply:
It'll lead to this error if params is not supplied
```
Too few arguments to function Enjin\Platform\Beam\Events\CreateBeamClaimsCompleted::__construct(), 1 passed in /Users/atudz/Documents/workspace/enjin-internal-platform/app/GraphQL/Queries/GetEventsQuery.php on line 229 and exactly 3 expected  
```

Reply:
PlatformBroadcastedEvent class handles that.... It defaults to null and pass the nulls to the constructor of the child class. (When calling safeBroadcast)

Is it not being handled?

Reply:
It's not working, I'm getting that exception above when only passing 1 parameter.

Reply:
That's the thing, you should not instantiate the event manualy, it should be called through safeBroadcast.

Reply:
I need to instantiate it manually to get the object instance and get its details. It's being used on internal platform
https://github.com/enjin/enjin-internal-platform/pull/185/files#diff-caa0ff5610635631a4670b843efd1a9b8ef5d6a5ff8538908fabbc0babb84d71R237

Reply:
No, it needs to get the actual class as I need to fetch the broadcastOn and broadcastWith data

Reply:
done